packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -0,0 +1,952 @@
+  });
+
+  input.dispatchEvent(new Event('change', { bubbles: true }));
+  input.dispatchEvent(new Event('input', { bubbles: true }));
+
+  return input;
+}
+
+const getPopup = (): NodeListOf<Element> => document.querySelectorAll('.dx-overlay-wrapper.dx-scheduler-appointment-popup');
+
+function openPopupByDblClick(POM: SchedulerModel, text?: string): AppointmentModel {
+  const appointment = POM.getAppointment(text) as AppointmentModel;
+
+  if (!appointment?.element) {
+    throw new Error(`Appointment "${text}" not found`);
+  }
+
+  appointment.element.click();
+  appointment.element.click();
+
+  return appointment;
+}
+
+function selectRadio(container: HTMLElement, value: string): Element | null {
+  const group = container.querySelector('[role="radiogroup"]');
+  if (!group) throw new Error('Radiogroup not found');
+
+  const radios = Array.from(group.querySelectorAll('[role="radio"]'));
+
+  const target = radios.find((radio) => {
+    const label = radio.getAttribute('aria-label')?.trim();
+    const text = radio.textContent?.trim();
+    return label === value || text === value;
+  });
+
+  if (!target) throw new Error(`Radio with value "${value}" not found`);
+
+  radios.forEach((r) => {
+    r.setAttribute('aria-checked', 'false');
+    r.classList.remove('dx-item-selected', 'dx-radiobutton-checked');
+  });
+
+  target.setAttribute('aria-checked', 'true');
+  target.classList.add('dx-item-selected', 'dx-radiobutton-checked');
+
+  target.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+
+  return target;
+}
+
+function getSelectedRadio(container: HTMLElement): HTMLElement | null {
+  return container.querySelector('[role="radio"][aria-checked="true"]');
+}
+
+function getSelectedRadioValue(container: HTMLElement): string | null {
+  const selected = getSelectedRadio(container);
+  return selected?.getAttribute('aria-label') ?? selected?.textContent?.trim() ?? null;
+}
+
+describe('Appointment popup form', () => {
+  beforeEach(() => {
+    fx.off = true;
+    setupSchedulerTestEnvironment();
+  });
+
+  afterEach(() => {
+    fx.off = false;
+    document.body.innerHTML = '';
+    jest.useRealTimers();
+  });
+
+  it('Original appointment\'s fields shouldn\'t fill if used fieldExpr', async () => {
+    const data: Record<string, unknown>[] = [];
+    const textExpValue = 'Subject';
+
+    let newAppointment: Record<string, unknown> | null = null;
+
+    const { scheduler, POM } = await createScheduler({
+      dataSource: data,
+      views: ['week'],
+      currentView: 'week',
+      currentDate: new Date(2021, 4, 27),
+      textExpr: textExpValue,
+      onAppointmentAdded: ({ appointmentData }) => {
+        newAppointment = appointmentData;
+      },
+      height: 600,
+    });
+
+    scheduler.showAppointmentPopup();
+
+    const form = POM.getForm() as HTMLElement;
+    setInputValueByLabel('Subject', 'qwerty', form);
+
+    POM.clickDoneButton();
```

Comment:
Better use object model instead of plain function.
For example:
`setInputValueByLabel('Subject', 'qwerty', form);`
`POM.clickDoneButton();`
=>
```
const popup = POM.getPopup();
popup.getInput('Subject').setValue('qwerty')
popup.getDoneButton().click()

```
It require more work with object model, but it's more readable